Generals/Code/GameEngine/Source/GameLogic/Map/SidesList.cpp

```diff
@@ -1124,20 +1124,18 @@ void TeamsInfoRec::addTeam(const Dict* d)
 	if (m_numTeams >= m_numTeamsAllocated)
 	{
 	// pool[]ify
-		TeamsInfo* nti = NEW TeamsInfo[m_numTeamsAllocated + TEAM_ALLOC_CHUNK];	// throws on failure
```

Comment:
`m_numTeamsAllocated + TEAM_ALLOC_CHUNK` would be unsafe if `m_numTeams` was larger than `m_numTeamsAllocated + TEAM_ALLOC_CHUNK`. I adjusted the logic for it but the compiler is still not happy with it.

Reply:
I checked again and the compiler is actually happy now. Probably it did not refresh the last time I looked.